regolith/helpers/makeappointmentshelper.py

```diff
@@ -0,0 +1,132 @@
+    subpi.add_argument("-a", "--appoint", action="store_true", help='suggest new appointments')
+
+    return subpi
+
+
+class MakeAppointmentsHelper(SoutHelperBase):
+    """Helper for managing appointments.
+
+       Appointments are...
+    """
+    # btype must be the same as helper target in helper.py
+    btype = HELPER_TARGET
+    needed_dbs = [f'{TARGET_COLL}', "grants"]
+
+    def construct_global_ctx(self):
+        """Constructs the global context"""
+        super().construct_global_ctx()
+        gtx = self.gtx
+        rc = self.rc
+        if "groups" in self.needed_dbs:
+            rc.pi_id = get_pi_id(rc)
+        rc.coll = f"{TARGET_COLL}"
+        try:
+            if not rc.database:
+                rc.database = rc.databases[0]["name"]
+        except:
+            pass
+        colls = [
+            sorted(
+                all_docs_from_collection(rc.client, collname), key=_id_key
+            )
+            for collname in self.needed_dbs
+        ]
+        for db, coll in zip(self.needed_dbs, colls):
+            gtx[db] = coll
+        gtx["all_docs_from_collection"] = all_docs_from_collection
+        gtx["float"] = float
+        gtx["str"] = str
+        gtx["zip"] = zip
+
+
+    def sout(self):
+        rc = self.rc
+        outdated, depleted, underspent, overspent = [], [], [], []
+        if rc.begin_date:
+            begin_date = date_parser.parse(rc.begin_date).date()
+        else:
+            begin_date = None
+        if rc.end_date:
+            end_date = date_parser.parse(rc.end_date).date()
+        else:
+            end_date = None
+
+        for p in self.gtx["people"]:
+            if p.get("appointments"):
+                appts = p.get("appointments")
+                is_fully_appointed(p, begin_date=begin_date, end_date=end_date)
+                for a in appts:
+                    g = rc.client.find_one(rc.database, self.gtx["grants"], {"_id": a.get("grant")})
+                    timespan = end_date - begin_date
+                    for x in range(0, timespan.days):
+                        day = begin_date + relativedelta(days=x)
+                        if not is_current(g, day):
+                            outdated.append(
+                            f"person: {[p].get('_id')} appointment: {a.get('_id')} grant: {g.get('_id')} "
+                            f"date: {str(day)}")
+                        if get_grant_amount(g, self.gtx["people"], day, day) <= 0:
+                        # define this in tools.py
+                            depleted.append(
+                                f"person: {[p].get('_id')} appointment: {a.get('_id')} grant: {g.get('_id')} "
+                                f"date: {str(day)}")
+
+        for g in self.gtx["grants"]:
+            end_date = get_dates(g)['end_date']
+            if get_grant_amount(g, self.gtx["people"], end_date, end_date) > 100:
+                # what is the required 'underspend' amount?
```

Comment:
this is just a crude inequality for the time being! what threshold for `student_months`, `postdoc_months` or `ss_months` should be used here?

Reply:
I took a different approach which was to plot the burn, but I like this functionality too.   Are the "units" student/postdoc months?  In that case probably an over or under of 3 months or more is about right.

Reply:
thank you! I can also work on plotting the burn – i think that would look like extracting the values from this list and mapping time vs sum of ss_months, student_months and postdoc_months. so I should check if `student_months + postdoc_months < 3` on the `end_date` of the grant, correct? do we not factor in `ss_months`?

Reply:
yes, that's right, though check that the sum at the end is less than 1, not 3.  We should also put in ss months.   The burn plot looks funny because it has vertical steps on the anniversary of each grant as n student_monthe etc. are added in each budget year, and then slopes down as the appointments eat it up one day at a time.

Reply:
done!